File: server/router/ExpertDatabase/expert_adoption_record.go
package ExpertDatabase

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type ExpertAdoptionRecordRouter struct{}

// InitExpertAdoptionRecordRouter 初始化 专家决策影响记录 路由信息
func (s *ExpertAdoptionRecordRouter) InitExpertAdoptionRecordRouter(Router *gin.RouterGroup) {
	expertAdoptionRecordRouter := Router.Group("expertAdoptionRecord").Use(middleware.OperationRecord())
	expertAdoptionRecordRouterWithoutRecord := Router.Group("expertAdoptionRecord")
	expertAdoptionRecordApi := v1.ApiGroupApp.ExpertDatabaseApiGroup.ExpertAdoptionRecordApi
	{
		expertAdoptionRecordRouter.POST("createExpertAdoptionRecord", expertAdoptionRecordApi.CreateExpertAdoptionRecord)             // 新建专家决策影响记录
		expertAdoptionRecordRouter.DELETE("deleteExpertAdoptionRecord", expertAdoptionRecordApi.DeleteExpertAdoptionRecord)           // 删除专家决策影响记录
		expertAdoptionRecordRouter.DELETE("deleteExpertAdoptionRecordByIds", expertAdoptionRecordApi.DeleteExpertAdoptionRecordByIds) // 批量删除专家决策影响记录
		expertAdoptionRecordRouter.PUT("updateExpertAdoptionRecord", expertAdoptionRecordApi.UpdateExpertAdoptionRecord)              // 更新专家决策影响记录
	}
	{
		expertAdoptionRecordRouterWithoutRecord.GET("findExpertAdoptionRecord", expertAdoptionRecordApi.FindExpertAdoptionRecord)       // 根据ID获取专家决策影响记录
		expertAdoptionRecordRouterWithoutRecord.GET("getExpertAdoptionRecordList", expertAdoptionRecordApi.GetExpertAdoptionRecordList) // 分页获取专家决策影响记录列表
	}
}

File: web/src/api/expertAdoptionRecord.js
import service from '@/utils/request'

// @Tags ExpertAdoptionRecord
// @Summary 创建专家决策影响记录
// @Router /expertAdoptionRecord/createExpertAdoptionRecord [post]
export const createExpertAdoptionRecord = (data) => {
  return service({
    url: '/expertAdoptionRecord/createExpertAdoptionRecord',
    method: 'post',
    data
  })
}

// @Tags ExpertAdoptionRecord
// @Summary 删除专家决策影响记录
// @Router /expertAdoptionRecord/deleteExpertAdoptionRecord [delete]
export const deleteExpertAdoptionRecord = (params) => {
  return service({
    url: '/expertAdoptionRecord/deleteExpertAdoptionRecord',
    method: 'delete',
    params
  })
}

// @Tags ExpertAdoptionRecord
// @Summary 批量删除专家决策影响记录
// @Router /expertAdoptionRecord/deleteExpertAdoptionRecordByIds [delete]
export const deleteExpertAdoptionRecordByIds = (params) => {
  return service({
    url: '/expertAdoptionRecord/deleteExpertAdoptionRecordByIds',
    method: 'delete',
    params
  })
}

// @Tags ExpertAdoptionRecord
// @Summary 更新专家决策影响记录
// @Router /expertAdoptionRecord/updateExpertAdoptionRecord [put]
export const updateExpertAdoptionRecord = (data) => {
  return service({
    url: '/expertAdoptionRecord/updateExpertAdoptionRecord',
    method: 'put',
    data
  })
}

// @Tags ExpertAdoptionRecord
// @Summary 用id查询专家决策影响记录
// @Router /expertAdoptionRecord/findExpertAdoptionRecord [get]
export const findExpertAdoptionRecord = (params) => {
  return service({
    url: '/expertAdoptionRecord/findExpertAdoptionRecord',
    method: 'get',
    params
  })
}

// @Tags ExpertAdoptionRecord
// @Summary 分页获取专家决策影响记录列表
// @Router /expertAdoptionRecord/getExpertAdoptionRecordList [get]
export const getExpertAdoptionRecordList = (params) => {
  return service({
    url: '/expertAdoptionRecord/getExpertAdoptionRecordList',
    method: 'get',
    params
  })
}
